Simplify and streamline the type inspection API (#1186)
actor-framework/actor-framework: manual/TypeInspection.rst

    }
  }

This clearly appears more verbose than it needs to be. Users generally need not
care about such internal types like ``id`` that only exist as a safeguard during
programming.

Hence, we generally recommend making such types transparent to CAF inspectors.
For our ``id`` type, the ``inspect`` overload may instead look as follows:

.. code-block:: C++

  template <class Inspector>
  bool inspect(Inspector& f, id& x) {
    return f.apply(x.value);
  }

> With this implementation instead of the previous one, inspectors simply read or

Reviewer: ```suggestion
In contrast to the previous implementation, inspectors now simply read or
```

> write strings whenever they encounter an ``id`` as value. This simplifies our

Reviewer: I'm not entirely sure what you want to say, but the "as value" confuses me. Maybe this?

```suggestion
write the strings as values whenever they encounter an ``id``. This simplifies our
```

gives the best performance. However, using the constant names results in a much
better user experience in all other cases.

The following code illustrates how to provide a string representation for

> inspectors that operate on human-readable data representation and the underlying
> type for an ``enum class`` otherwise.

Reviewer: ```suggestion
inspectors that operate on a human-readable data representation. Other inspectors
continue to use the underlying type of the ``enum class``.
```
